### ISSUE TITLE
Argo EventSource gives error during webhook creation in enterprise github

### ISSUE THREAD

**Issue by avinashkumar1289** [argoproj/argo-events#1005]
**Describe the bug**
While trying to create a webhook for enterprise Gitub argo events pod gives 406 error code

**To Reproduce**
* Create the event source for github
* Eventsource yaml file
```
# Info on GitHub Webhook: https://developer.github.com/v3/repos/hooks/#create-a-hook
apiVersion: argoproj.io/v1alpha1
kind: EventSource
metadata:
  name: github
spec:
  service:
    ports:
      - port: 12000
        targetPort: 12000
  github:
    example:
      # owner of the repo
      owner: avinashkumar-j
      # repository name
      repository: kubernetes/poc-catalog-dev
      # Github will send events to following port and endpoint
      webhook:
        # endpoint to listen to events on
        endpoint: /push
        # port to run internal HTTP server on
        port: "12000"
        # HTTP request method to allow. In this case, only POST requests are accepted
        method: POST
        # url the event-source will use to register at Github.
        # This url must be reachable from outside the cluster.
        # The name for the service is in `<event-source-name>-eventsource-svc` format.
        # You will need to create an Ingress or Openshift Route for the event-source service so that it can be reached from GitHub.
        url: http://xyz
      # type of events to listen to.
      # following listens to everything, hence *
      # You can find more info on https://developer.github.com/v3/activity/events/types/
      events:
        - "*"

      githubBaseURL: https://github01.custom.com/

      # apiToken refers to K8s secret that stores the github api token
      # if apiToken is provided controller will create webhook on GitHub repo
      # +optional
      apiToken:
        # Name of the K8s secret that contains the access token
        name: github-access
        # Key within the K8s secret whose corresponding value (must be base64 encoded) is access token
        key: token

#      # webhookSecret refers to K8s secret that stores the github hook secret
#      # +optional
#      webhookSecret:
#        # Name of the K8s secret that contains the hook secret
#        name: github-access
#        # Key within the K8s secret whose corresponding value (must be base64 encoded) is hook secret
#        key: secret

      # type of the connection between event-source and Github.
      # You should set it to false to avoid man-in-the-middle and other attacks.
      insecure: false
      # Determines if notifications are sent when the webhook is triggered
      active: true
      # The media type used to serialize the payloads
      contentType: json
```

**Expected behavior**
The webhook should have been created.

**Error message**
```
"error": "failed to create webhook. err: GET https://github01.custom.com/login?return_to=https%3A%2F%2Fgithub01.custom.com%2Frepos%2Favinashkumar-j%2Fkubernetes%2Fpoc-catalog-dev%2Fhooks: 406  []",
	"errorVerbose": "failed to create webhook. err: GET https://github01.custom.com/login?return_to=https%3A%2F%2Fgithub01.custom.com%2Frepos%2Favinashkumar-j%2Fkubernetes%2Fpoc-catalog-dev%2Fhooks: 406  []\ngithub.com/argoproj/argo-events/eventsources/sources/github.(*Router).PostActivate\n\t/home/runner/work/argo-events/argo-events/eventsources/sources/github/start.go:217\ngithub.com/argoproj/argo-events/eventsources/common/webhook.ManageRoute\n\t/home/runner/work/argo-events/argo-events/eventsources/common/webhook/webhook.go:231\ngithub.com/argoproj/argo-events/eventsources/sources/github.(*EventListener).StartListening\n\t/home/runner/work/argo-events/argo-events/eventsources/sources/github/start.go:282\ngithub.com/argoproj/argo-events/eventsources.(*EventSourceAdaptor).Start.func2\n\t/home/runner/work/argo-events/argo-events/eventsources/eventing.go:306\nruntime.goexit\n\t/opt/hostedtoolcache/go/1.14.7/x64/src/runtime/asm_amd64.s:1373",
	"stacktrace": "github.com/argoproj/argo-events/eventsources.(*EventSourceAdaptor).Start.func2\n\t/home/runner/work/argo-events/argo-events/eventsources/eventing.go:327"
```

**Environment:**
 - Kubernetes: [e.g. v1.15.12]
 - Argo Events: [e.g. v1.0.0]

**Additional context**

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by avinashkumar1289** [comment]
The issue is fixed

**Comment by cal-pratt** [comment]
> The issue is fixed

@avinashkumar1289 How has this been fixed? I'm experiencing this issue using `v1.3.0-rc1`

```
apiVersion: argoproj.io/v1alpha1
kind: EventSource
metadata:
  name: github
spec:
  service:
    ports:
      - port: 12000
        targetPort: 12000
  github:
    mycustomrepo:
      githubBaseURL: https://github.myorg.com/
      owner: myuser
      repository: mycustomrepo
      webhook:
        endpoint: /myuser/mycustomrepo
        port: "12000"
        method: POST
        url: http://github-eventsource.argo.myorg.com/
      events:
        - "*"
      insecure: true
      active: true
      contentType: json
      apiToken:
        name: github-access
        key: token
```

```
{
  "level": "error",
  "ts": 1617495433.4875581,
  "logger": "argo-events.eventsource",
  "caller": "webhook/webhook.go:234",
  "msg": "error occurred while performing post route activation operations",
  "eventSourceName": "github",
  "eventSourceType": "github",
  "eventName": "mycustomrepo",
  "error": "failed to create webhook. err: GET https://github.myorg.com/login?return_to=https%3A%2F%2Fgithub.myorg.com%2Frepos%2Fmyuser%2Fmycustomrepo%2Fhooks: 406  []",
  "errorVerbose": "failed to create webhook. err: GET https://github.myorg.com/login?return_to=https%3A%2F%2Fgithub.myorg.com%2Frepos%2Fmyuser%2Fmycustomrepo%2Fhooks: 406  []\ngithub.com/argoproj/argo-events/eventsources/sources/github.(*Router).PostActivate\n\t/home/runner/work/argo-events/argo-events/eventsources/sources/github/start.go:225\ngithub.com/argoproj/argo-events/eventsources/common/webhook.ManageRoute\n\t/home/runner/work/argo-events/argo-events/eventsources/common/webhook/webhook.go:233\ngithub.com/argoproj/argo-events/eventsources/sources/github.(*EventListener).StartListening\n\t/home/runner/work/argo-events/argo-events/eventsources/sources/github/start.go:299\ngithub.com/argoproj/argo-events/eventsources.(*EventSourceAdaptor).Start.func3.1\n\t/home/runner/work/argo-events/argo-events/eventsources/eventing.go:345\ngithub.com/argoproj/argo-events/common.Connect.func1\n\t/home/runner/work/argo-events/argo-events/common/retry.go:98\nk8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtection\n\t/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.19.7-rc.0/pkg/util/wait/wait.go:211\nk8s.io/apimachinery/pkg/util/wait.ExponentialBackoff\n\t/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.19.7-rc.0/pkg/util/wait/wait.go:399\ngithub.com/argoproj/argo-events/common.Connect\n\t/home/runner/work/argo-events/argo-events/common/retry.go:97\ngithub.com/argoproj/argo-events/eventsources.(*EventSourceAdaptor).Start.func3\n\t/home/runner/work/argo-events/argo-events/eventsources/eventing.go:344\nruntime.goexit\n\t/opt/hostedtoolcache/go/1.14.15/x64/src/runtime/asm_amd64.s:1373",
  "stacktrace": "github.com/argoproj/argo-events/eventsources/common/webhook.ManageRoute\n\t/home/runner/work/argo-events/argo-events/eventsources/common/webhook/webhook.go:234\ngithub.com/argoproj/argo-events/eventsources/sources/github.(*EventListener).StartListening\n\t/home/runner/work/argo-events/argo-events/eventsources/sources/github/start.go:299\ngithub.com/argoproj/argo-events/eventsources.(*EventSourceAdaptor).Start.func3.1\n\t/home/runner/work/argo-events/argo-events/eventsources/eventing.go:345\ngithub.com/argoproj/argo-events/common.Connect.func1\n\t/home/runner/work/argo-events/argo-events/common/retry.go:98\nk8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtection\n\t/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.19.7-rc.0/pkg/util/wait/wait.go:211\nk8s.io/apimachinery/pkg/util/wait.ExponentialBackoff\n\t/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.19.7-rc.0/pkg/util/wait/wait.go:399\ngithub.com/argoproj/argo-events/common.Connect\n\t/home/runner/work/argo-events/argo-events/common/retry.go:97\ngithub.com/argoproj/argo-events/eventsources.(*EventSourceAdaptor).Start.func3\n\t/home/runner/work/argo-events/argo-events/eventsources/eventing.go:344"
}
```

**Comment by cal-pratt** [comment]
Nevermind. I figured it out.  You have to make sure that the `githubBaseURL` includes any path prefix you have to the API endpoints. In my case I need to the set endpoint to:

```yaml
githubBaseURL: https://github.myorg.com/api/v3/
```

Might be good to have this in the docs somewhere for other dummies like me :)
